gym-unity/gym_unity/envs/__init__.py

```diff
@@ -432,7 +453,11 @@ def _step(self, needs_reset: bool = False) -> BatchedStepResult:
                     "The environment does not have the expected amount of agents."
                     + "Some agents did not request decisions at the same time."
                 )
-            self._done_agents.update(list(info.agent_id))
+            for agent_id, reward in zip(info.agent_id, info.reward):
```

Comment:
Same comment about building the reverse lookup here.

Reply:
I do not think the same trick will work here because there can me multiple dead agents at once (ie, with a -1 agent_id)